contracts/contracts/processors/BaseProcessor.sol

```diff
@@ -68,4 +68,29 @@ contract BaseProcessor is Ownable {
         require(!nullifierRegistry.isNullified(_nullifier), "Nullifier has already been used");
         nullifierRegistry.addNullifier(_nullifier);
     }
+
+    // stringToUint is used to convert a string like "45" to a uint256 4
```

Comment:
Fix the comment at the top of _stringToUint. Add verbose cases. Example, for expected decimals 2, 12.34 -> 1234, 12.3 -> 1230, 12 -> 1200, 12.345 -> reverts.